src/partition/repart.c

```diff
@@ -2455,6 +2455,18 @@ static int partition_read_definition(Partition *p, const char *path, const char
                     PARTITION_USR_VERITY) || p->verity == VERITY_DATA) && p->read_only < 0)
                 p->read_only = true;
 
+        /* Don't automatically grow ESP, XBOOTLDR and verity signature partitions.*/
+        if (p->weight == UINT32_MAX) {
+                if (IN_SET(p->type.designator,
+                           PARTITION_ROOT_VERITY_SIG,
+                           PARTITION_USR_VERITY_SIG,
+                           PARTITION_ESP,
+                           PARTITION_XBOOTLDR))
+                        p->weight = 0;
+                else
+                        p->weight = 1000;
```

Comment:
hmm, is this really the right approach? I mean, i can see how it makes little sense to increase the size of an existing ESP/XBOOTLDR (because we cannot resonably grow vfat at this time). But if we create a fresh disk image from scratch, it sounds OK to size things based on weights, no?

i mean for ESP sizemin/sizemax also is set to undefined. so if we have no real weight and no size constraints, then what would we even size the partition to?